### PR TITLE
Lease: Implement keepAliveOnce

### PR DESCRIPTION
**Pull request by fanminshi**

### CODE REVIEW

File: src/test/java/com/coreos/jetcd/LeaseTest.java [hunk 1]
```diff
@@ -20,45 +21,58 @@
...
-    test.assertEquals(kvClient.get(testKey).get().getCount(), 0);
+    test.assertEquals(kvClient.get(KEY).get().getCount(), 0);
   }
 
   @Test(dependsOnMethods = "testRevoke")
+  public void testKeepAliveOnce() throws ExecutionException, InterruptedException {
+    long leaseID = leaseClient.grant(2).get().getID();
+    kvClient.put(KEY, VALUE, PutOption.newBuilder().withLeaseId(leaseID).build()).get();
+    test.assertEquals(kvClient.get(KEY).get().getCount(), 1);
+    Thread.sleep(1000);
```
Comment (xiang90) [inline src/test/java/com/coreos/jetcd/LeaseTest.java]: why do we need this sleep?

File: src/main/java/com/coreos/jetcd/LeaseImpl.java [hunk 2]
```diff
@@ -216,16 +216,31 @@ public synchronized void cancelKeepAlive(long leaseId)
...
+          @Override
+          public void onCompleted() {
+          }
+        });
     requestObserver.onNext(newKeepAliveRequest(leaseId));
-    requestObserver.onCompleted();
 
-    throw new UnsupportedOperationException();
+    // cancel grpc stream when leaseKeepAliveResponseCompletableFuture completes.
+    // TODO: manage Async threads.
```
Comment (xiang90) [inline src/main/java/com/coreos/jetcd/LeaseImpl.java]: what do you mean by managing Async threads?

File: src/test/java/com/coreos/jetcd/LeaseTest.java [hunk 3]
```diff
@@ -20,45 +21,58 @@
...
+    leaseClient.keepAliveOnce(leaseID).get();
```
Comment (xiang90) [inline src/test/java/com/coreos/jetcd/LeaseTest.java]: check keep alive response here.

File: src/test/java/com/coreos/jetcd/LeaseTest.java [hunk 4]
```diff
@@ -20,45 +21,58 @@
...
+    Thread.sleep(1500);
+    test.assertEquals(kvClient.get(KEY).get().getCount(), 1);
```
Comment (xiang90) [inline src/test/java/com/coreos/jetcd/LeaseTest.java]: remove 64-67. when we know server returns good response, we know keep alive request sent out correctly. we do not need to test server behavior in client test.

**Comment by xiang90** [comment]
lgtm

File: src/main/java/com/coreos/jetcd/LeaseImpl.java (hunk 2 above)
Comment (fanminshi) [inline src/main/java/com/coreos/jetcd/LeaseImpl.java]: I think we should have an internal executor to handle the async threads. So when we close the client, we know where all our threads all. By default, the whenCompleteAsync without a exectuor use a common thread pool. 

"All async methods without an explicit Executor argument are performed using the ForkJoinPool.commonPool()" 
https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html

File: src/test/java/com/coreos/jetcd/LeaseTest.java (hunk 4 above)
Comment (fanminshi) [inline src/test/java/com/coreos/jetcd/LeaseTest.java]: sounds good.

File: src/test/java/com/coreos/jetcd/LeaseTest.java (hunk 1 above)
Comment (fanminshi) [inline src/test/java/com/coreos/jetcd/LeaseTest.java]: probably don't need. we can just check  the  keepalive response instead.

File: src/main/java/com/coreos/jetcd/LeaseImpl.java (hunk 2 above)
Comment (xiang90) [inline src/main/java/com/coreos/jetcd/LeaseImpl.java]: ok, makes sense. need a better comment.

**Pull request merged**